global.gd

```diff
@@ -32,17 +54,31 @@ func readFromJSON(path: String) -> Dictionary:
 		printerr("Invalid path given")
 		return data
 
-func saveToJSON(path: String, data: Dictionary) -> void:
+func getItemByKey(key: String, json: Dictionary) -> Dictionary:
+	if json and json.has(key):
+		return json[key]
+	else:
+		var a: Dictionary
+		return a
+
+
+func saveToJSON(path: String, SaveData: Dictionary) -> void:
 	var file := File.new()
 	file.open(path, File.WRITE)
-	var jsonText = JSON.print(data)  # Use JSON.print to convert dictionary to JSON
+	var jsonText = JSON.print(SaveData)  # Use JSON.print to convert dictionary to JSON
 	file.store_string(jsonText)
 	file.close()
 
-func getItemByKey(key: String, path: String) -> Dictionary:
-	var json = readFromJSON(path)
-	if json and json.has(key):
-		return items[key].duplicate(true)
-	else:
-		var a: Dictionary
-		return a
+func appendToJSON(path: String, newData: Dictionary) -> void:
+	# Read the existing JSON data
+	var existingData = readFromJSON(path)
+
+	# Merge the existing data with the new data
+	for key in newData.keys():
+		existingData[key] = newData[key]
+
+	# Save the updated JSON data
+	saveToJSON(path, existingData)
+
+func setInv(key: String, property: String, newProperty) -> void:
+	setInfo(key, property, newProperty, global.inventory, global.inventoryPath)
```

Comment:
New functions:
appendToJSON():
Instead of taking in a dictionary and then overwriting the existing JSON file with it(which is what the setInfo() did), and not being able to create new keys for new objects automatically, appendToJSON() adds a new key if it doesn't already exist, and if it does, updates it and then overwrites it over the JSON,
setInv():
Instead of having to specify all of the arguments required for setInfo(), which is mostly the same when it comes to changing values for the inventory JSON, we just have this simple function that takes care of that for us,